src/alert.cpp

```diff
@@ -1855,7 +1855,7 @@ namespace libtorrent {
 
 	char const* dht_pkt_alert::pkt_buf() const
 	{
-		return m_alloc.ptr(m_msg_idx);
+		return m_size > 0 ? m_alloc.ptr(m_msg_idx) : nullptr;
```

Comment:
instead of putting this in 3 places, you could just put it in the ptr() function